packages/tailwindcss/src/ast.ts

```diff
@@ -137,39 +137,54 @@ export function walk(
     }
 
     path.push(node)
+    let replacedNode = false
```

Comment:
We probably want to port this behavior to other `walk` implementations in the codebase:

https://github.com/tailwindlabs/tailwindcss/blob/d566dbdef658b8c98a8c596466d74f11c72295e2/packages/tailwindcss/src/compat/selector-parser.ts#L82-L130

and 

https://github.com/tailwindlabs/tailwindcss/blob/d566dbdef658b8c98a8c596466d74f11c72295e2/packages/tailwindcss/src/value-parser.ts#L53-L101

Reply:
ay